shadowsocks-csharp/Controller/Service/Sip003Plugin.cs

```diff
@@ -94,7 +94,18 @@ public bool StartIfNeeded()
                 _pluginProcess.StartInfo.Environment["SS_LOCAL_HOST"] = LocalEndPoint.Address.ToString();
                 _pluginProcess.StartInfo.Environment["SS_LOCAL_PORT"] = LocalEndPoint.Port.ToString();
                 _pluginProcess.StartInfo.Arguments = ExpandEnvironmentVariables(_pluginProcess.StartInfo.Arguments, _pluginProcess.StartInfo.EnvironmentVariables);
-                _pluginProcess.Start();
+                try
+                {
+                    _pluginProcess.Start();
+                }
+                catch (System.ComponentModel.Win32Exception ex)
+                {
+                    // do not use File.Exists(...), it can not handle the scenarios when the plugin file is in system environment path.
+                    if ((uint)ex.ErrorCode == 0x80004005)   // file not found

```

Comment:
FYI: #2729

Reply:
Thanks, will use native error code to find out the real problem
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d

Reply:
But it is uncommon to see "The specified executable is not a valid application for this OS platform." error. Only when run 64 bit program on 32 bit platform. or just have the bad image.

Reply:
Yes, it's uncommon, but there is😥. Just check if it is FileNotFound, if not, throw error with (or without) native error code so we won't misleading user.

Reply:
Personally prefer to use current implementation to keep everything simple.

Reply:
Copy paste from here
https://docs.microsoft.com/zh-cn/windows/win32/seccrypto/common-hresult-values?redirectedfrom=MSDN


Name | Description | Value
-- | -- | --
S_OK | Operation successful | 0x00000000
E_ABORT | Operation aborted | 0x80004004
E_ACCESSDENIED | General access denied error | 0x80070005
E_FAIL | Unspecified failure | 0x80004005
......|......|.......

Seems we should use native error code...